### PR TITLE
Automated CLI Scap-content tests [6.3 Feature]

### PR DESCRIPTION
**Pull request by san7ket**
```
 pytest tests/foreman/cli/test_oscap.py
 test session starts 
platform linux2 -- Python 2.7.12, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /home/sjagtap/PycharmProjects/robottelo, inifile:

tests/foreman/cli/test_oscap.py s..ss.s.sssssssssssssss.sssssssss

 5 passed, 28 skipped in 117.93 seconds 
```

### CODE REVIEW

File: tests/foreman/cli/test_oscap.py [hunk 1]
```diff
@@ -24,28 +35,41 @@
...
-    """Returns a list of valid environment names"""
-    return[
+def invalid_id_list():
+    """Returns a list of invalid ids"""
+    return [
         gen_string('alpha'),
-        gen_string('numeric'),
-        gen_string('alphanumeric'),
-        gen_string('alphanumeric', 255)
+        gen_string('alphanumeric')
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: Could you please describe these changes? Numeric values aren't supported anymore? 255 chars is not valid boundary value? What's the new one then?
Btw, pls also think of moving this helper to datafactory, i'm not sure why it's not there yet.

File: tests/foreman/cli/test_oscap.py [hunk 2]
```diff
@@ -24,28 +35,41 @@
...
     ]
 
 
 class OpenScapTestCase(CLITestCase):
     """Tests related to the oscap cli hammer plugin"""
 
     @classmethod
+    def create_test_user_viewer_role(cls):
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: Docstring is required for all the methods

File: tests/foreman/cli/test_oscap.py [hunk 3]
```diff
@@ -24,28 +35,41 @@
...
+        cls.login = gen_string('alpha')
+        cls.password = gen_string('alpha')
+        user = make_user({
+            'login': cls.login,
+            'password': cls.password,
+            'admin': '0'
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: imo True/False looks better that string with numeric value, i'd rather used `'admin': False` here. Not even worth a nitpick, just fyi :)

File: tests/foreman/cli/test_oscap.py [hunk 4]
```diff
@@ -24,28 +35,41 @@
...
+        })
+        role = Role.info({
+            'name': 'Viewer'
+        })
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: Fits in one line:
```python
        role = Role.info({'name': 'Viewer'})
```

File: tests/foreman/cli/test_oscap.py [hunk 5]
```diff
@@ -75,13 +98,14 @@ def test_positive_list_default_content_with_admin(self):
 
         :expectedresults: The scap-content are listed.
 
-        :caseautomation: notautomated
+        :caseautomation: automated
 
         :CaseImportance: Critical
         """
+        result = Scapcontent.list()
+        self.assertGreater(len(result), 0)
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: Is there any default name/value, presence of which we can check? I'm afraid we can generate false positive here when empty string is returned - list with 1 item (empty string) will pass this check.

File: tests/foreman/cli/test_oscap.py [hunk 6]
```diff
@@ -128,13 +155,19 @@ def test_positive_view_scap_content_info_admin(self):
...
         :expectedresults: The info of the scap-content is listed.
 
-        :caseautomation: notautomated
+        :caseautomation: automated
 
         :CaseImportance: Critical
         """
+        title = gen_string('alpha')
+        make_scapcontent({'title': title, 'scap-file': '/tmp/{0}'.format(
+                            self.file_name)})
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: nitpick: If 2 dict keys do not fit in one line just use different style with 1 item per line:
```python
        make_scapcontent({
            'title': title,
            'scap-file': '/tmp/{0}'.format(self.file_name),
        })
```
It's much more readable.

File: tests/foreman/cli/test_oscap.py [hunk 7]
```diff
@@ -155,13 +188,18 @@ def test_negative_info_scap_content_viewer_role(self):
...
-        :caseautomation: notautomated
+        :caseautomation: automated
 
         :CaseImportance: Critical
         """
+        title = gen_string('alpha')
+        make_scapcontent({'title': title, 'scap-file': '/tmp/{0}'.format(
+                                           self.file_name)})
+        login, password = self.create_test_user_viewer_role()
+        result = Scapcontent.with_user(login, password).info({'name': title})
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: Should be `.info({'title': title})` here, shouldn't it?

File: tests/foreman/cli/test_oscap.py [hunk 8]
```diff
@@ -182,10 +220,14 @@ def test_negative_info_scap_content(self):
 
         :expectedresults: The info of the scap-content is not listed.
 
-        :caseautomation: notautomated
+        :caseautomation: automated
 
         :CaseImportance: Critical
         """
+        for id in invalid_id_list():
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: `id` is reserved for builtin `id()` func, pls use different name for variable.

File: tests/foreman/cli/test_oscap.py (hunk 6 above)
Comment (abalakh) [reply tests/foreman/cli/test_oscap.py]: Applicable for other tests as well

File: tests/foreman/cli/test_oscap.py (hunk 1 above)
Comment (san7ket) [reply tests/foreman/cli/test_oscap.py]: @abalakh these are invalid ID , which are supposed to be numeric

File: tests/foreman/cli/test_oscap.py (hunk 7 above)
Comment (san7ket) [reply tests/foreman/cli/test_oscap.py]: This was a bug , fixed in latest snap.

**Comment by san7ket** [comment]
```
 pytest tests/foreman/cli/test_oscap.py

2017-09-11 14:59:44 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings

tests/foreman/cli/test_oscap.py ...ss...sss..ss..ssssss.sFsssss.s

 FAILURES OpenScapTestCase.test_positive_update_scap_content_with_newtitle 
```
filled the bug and skipped the test

```
pytest tests/foreman/cli/test_oscap.py -k test_positive_update_scap_content_with_newtitle
 test session starts 

tests/foreman/cli/test_oscap.py s
============================ 32 tests deselected 1 skipped, 32 deselected in 3.75 seconds 
```

**Comment by san7ket** [comment]
@abalakh revisit all comments addressed

File: tests/foreman/cli/test_oscap.py [hunk 9]
```diff
@@ -75,13 +87,17 @@ def test_positive_list_default_content_with_admin(self):
...
-        :caseautomation: notautomated
+        :caseautomation: automated
 
         :CaseImportance: Critical
         """
+        result = Scapcontent.list()
+        scap_titles = []
+        for title in result:
+            scap_titles.append(str(title['title']))
+        self.assertIn(OSCAP_DEFAULT_CONTENT['rhel7_content'], scap_titles)
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: nitpick: If i got your idea right, this test can be rewritten like that:
```python
result = Scapcontent.list()
self.assertIn(
    OSCAP_DEFAULT_CONTENT['rhel7_content'],
    [str(scap['title']) for scap in result]
)
```

File: tests/foreman/cli/test_oscap.py [hunk 10]
```diff
@@ -13,9 +13,20 @@
...
+from robottelo.constants import OSCAP_DEFAULT_CONTENT
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.factory import CLIFactoryError
+from robottelo.cli.factory import make_scapcontent, make_user
+from robottelo.cli.role import Role
+from robottelo.cli.scapcontent import Scapcontent
+from robottelo.cli.user import User
 from robottelo.config import settings
+from robottelo.datafactory import (
+    valid_data_list,
```
Comment (abalakh) [inline tests/foreman/cli/test_oscap.py]: nitpick: pls put in alphabetical order

File: tests/foreman/cli/test_oscap.py [hunk 11]
```diff
@@ -24,38 +35,39 @@
...
+        })
+        return cls.login, cls.password
+
+    @classmethod
     def setUpClass(cls):
         super(OpenScapTestCase, cls).setUpClass()
         file_name = settings.oscap.content_path
         cls.file_name = file_name.split('/')[
-                        (file_name.split('/')).__len__() - 1]
+                     (file_name.split('/')).__len__() - 1]
```
Comment (ldjebran) [inline tests/foreman/cli/test_oscap.py]: settings.oscap.content_path is a file why not use
```python
cls.file_name = os.path.basename(settings.oscap.content_path)
```

File: tests/foreman/cli/test_oscap.py [hunk 12]
```diff
@@ -128,13 +146,18 @@ def test_positive_view_scap_content_info_admin(self):
...
 
-        :caseautomation: notautomated
+        :caseautomation: automated
 
         :CaseImportance: Critical
         """
+        title = gen_string('alpha')
+        make_scapcontent({
+            'title': title,
+            'scap-file': '/tmp/{0}'.format(self.file_name)})
```
Comment (ldjebran) [inline tests/foreman/cli/test_oscap.py]:
```python
'scap-file': os.path.join('/tmp', self.file_name)})
```

File: tests/foreman/cli/test_oscap.py [hunk 13]
```diff
@@ -269,13 +304,19 @@ def test_positive_create_scap_content_with_valid_originalfile_name(self):
 
         :expectedresults: The scap-content is not created.
```
Comment (ldjebran) [inline tests/foreman/cli/test_oscap.py]: @san7ket is there an error here?

Comment (san7ket) [reply tests/foreman/cli/test_oscap.py]: @ldjebran my bad, typo :/ , thanks for catching it

**Comment by san7ket** [comment]
@ldjebran please revisit, comments addressed
